connection.go

```diff
@@ -196,6 +196,10 @@ type Connection struct {
 	healthCheckQuit    context.CancelFunc
 	healthCheckDone    chan struct{}
 	healthCheckHistory *healthHistory
+
+	// lastActivity is used to track how long the connection has been idle.
+	// (unix time, nano)
+	lastActivity *atomic.Int64
```

Comment:
no reason this has to be a pointer, can you do atomic.Int64 here, and dereference when you create it using atomic.NewInt64

Reply:
Done